File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/SearchHistory.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "search_history")
public class SearchHistory implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "sh_id")
    private Long shId;

    @Column(name = "sh_date", nullable = false)
    private java.sql.Timestamp shDate;

    @Column(name = "sh_search", nullable = false)
    private String shSearch;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false, nullable = false)
    private User user;
}

File: Frontend/manga-world/src/Model/SearchHistory.ts
import User from "./user";

class searchHistory {
    shId: number;
    shDate: Date;
    shSearch: string;
    userId: number;
    user: User;

    constructor(shId: number, shDate: Date, shSearch: string, userId: number, user: User) {
        this.shId = shId;
        this.shDate = shDate;
        this.shSearch = shSearch;
        this.userId = userId;
        this.user = user;
    }
}

export default searchHistory;